core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -307,9 +307,13 @@ class KafkaApis(val requestChannel: RequestChannel,
             ErrorMapping.exceptionNameFor(status.error)))
           errorInResponse = true
         }
+
+        // When this callback is triggered, the remote API call has completed
+        request.apiRemoteCompleteTimeMs = SystemTime.milliseconds
```

Comment:
You probably want to have this just before the call to quota (and not in the foreach)